src/Microsoft.TemplateEngine.Edge/DefaultPathInfo.cs

```diff
@@ -97,9 +97,24 @@ public sealed class DefaultPathInfo : IPathInfo
 
         private static string GetUserProfileDir (IEnvironment environment)
         {
-            bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
-            return environment.GetEnvironmentVariable(isWindows ? "USERPROFILE" : "HOME")
-                ?? throw new NotSupportedException("HOME or USERPROFILE environment variable is not defined, the environment is not supported");
+            const string DotnetHomeVariableName = "DOTNET_CLI_HOME";
```

Comment:
Unfortunately we cannot use `DOTNET_CLI_HOME` here as it is specific to dotnet CLI, and `Edge` is used by other parties too.

The way forward may be to add another parameter for `DefaultPathInfo` that will specify the overriden user folder location. If it's not set, the default (previous version) may be used. `TemplateEngine.CLI` should check if `DOTNET_CLI_HOME` is set and if so, when creating `EngineIEnvironmentSettings` here https://github.com/dotnet/templating/blob/2851334289e12782b5c9b0264301a9c9bfc0af28/src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs#L50 pass the `DefaultPathInfo` with that path set.

Reply:
Or do completely separate `IDefaultPathInfo` implementation for `TemplateEngine.CLI`.
Unsealing this class and allow overriding the properties might be another option too to reuse the code for other properties.

Reply:
Thanks. I didn't realize the other use-cases of Edge engine. I have added a flag in `EngineEnvironmentSettings`'s ctor which indicates whether or not `DOTNET_CLI_HOME` should be probed. A new test case in included which ensures it is working as expected.

Reply:
We should avoid using `DOTNET_CLI_HOME` at all in `Edge`.

Please do evaluation of `DOTNET_CLI_HOME` in `TemplateEngine.CLI` and pass resolved user folder to `DefaultPathInfo` (via new parameter) or inherit from `DefaultPathInfo`  in `TemplateEngine.CLI` or do own implementation of `IDefaultPathInfo`  in `TemplateEngine.CLI`.

The only change in Edge should be `DefaultPathInfo` accepting custom path for user folder directory or its unsealing.

Reply:
Thanks, I have modified the implementation so `Edge` does not use `DOTNET_CLI_HOME`.